dataland-frontend/tests/e2e/fixtures/lksg/LksgDataFixtures.ts

```diff
@@ -7,534 +7,371 @@ import {
-        seasonalOrMigrantWorkers: valueOrUndefined(randomYesNo(), undefinedProbability),
-        shareOfTemporaryWorkers: valueOrUndefined(randomShareOfTemporaryWorkersInterval(), undefinedProbability),
-        annualTotalRevenue: valueOrUndefined(randomFloat(1000000, 10000000000, 1), undefinedProbability),
-        totalRevenueCurrency: valueOrUndefined(generateIso4217CurrencyCode(), undefinedProbability),
-        fixedAndWorkingCapital: valueOrUndefined(randomInt(10000), undefinedProbability),
+        dataDate: generateFutureDate(),
+        headOfficeInGermany: dataGenerator.randomYesNo(),
+        groupOfCompanies: dataGenerator.randomYesNo(),
+        groupOfCompaniesName: dataGenerator.valueOrUndefined(faker.company.name()),
+        industry: dataGenerator.valueOrUndefined(generateListOfRandomNaceCodes()),
+        numberOfEmployees: dataGenerator.valueOrUndefined(generateInt(10000)),
+        seasonalOrMigrantWorkers: dataGenerator.randomYesNo(),
+        shareOfTemporaryWorkers: dataGenerator.randomShareOfTemporaryWorkersInterval(),
+        annualTotalRevenue: dataGenerator.randomCurrencyValue(),
+        totalRevenueCurrency: dataGenerator.valueOrUndefined(generateIso4217CurrencyCode()),
+        fixedAndWorkingCapital: dataGenerator.valueOrUndefined(generateInt(10000)),
       },
       productionSpecific: {
-        manufacturingCompany: valueOrUndefined(randomYesNo(), undefinedProbability),
-        capacity: valueOrUndefined(
-          randomInt(25).toString() + " " + faker.commerce.product() + " per " + faker.date.weekday(),
-          undefinedProbability,
-        ),
-        productionViaSubcontracting: valueOrUndefined(randomYesNo(), undefinedProbability),
-        subcontractingCompaniesCountries: valueOrUndefined(
-          generateArray(getRandomIso2CountryCode),
-          undefinedProbability,
-        ),
-        subcontractingCompaniesIndustries: valueOrUndefined(generateListOfRandomNaceCodes(), undefinedProbability),
-        productionSites: valueOrUndefined(randomYesNo(), undefinedProbability),
-        listOfProductionSites: valueOrUndefined(
-          generateArray(() => generateProductionSite(undefinedProbability)),
-          undefinedProbability,
-        ),
-        market: valueOrUndefined(randomNationalOrInternationalMarket(), undefinedProbability),
-        specificProcurement: valueOrUndefined(randomYesNo(), undefinedProbability),
+        manufacturingCompany: dataGenerator.randomYesNo(),
+        capacity: dataGenerator.valueOrUndefined(
+          generateInt(25).toString() + " " + faker.commerce.product() + " per " + faker.date.weekday(),
+        ),
+        productionViaSubcontracting: dataGenerator.randomYesNo(),
+        subcontractingCompaniesCountries: dataGenerator.valueOrUndefined(generateArray(generateIso2CountryCode)),
+        subcontractingCompaniesIndustries: dataGenerator.valueOrUndefined(generateListOfRandomNaceCodes()),
```

Comment:
data generator has a randomArray mehtod

Reply:
changed

Reply:
The field `subcontractingCompaniesIndustries` is still using a `valueOrUndefined` wrapper.
It could be replaced by writing `dataGenerator.randomArray(generateListOfRandomNaceCodes())`,

Reply:
(The field `industry` in the `general` segment too)

Reply:
I do not think this is a good idea. The function generateNaceCodes returns already a list of NACE codes and wrapping this into valueOrUndefined directly yields the desired result. If you use randomArray it requires a generator as an input. If you use generateNaceCodes as a generator you will end up with an array of arrays of NACE codes. You would need a generator that returns one single NACE code to use it in that fashion. Hence, I would just leave it.

Reply:
Ah of course, you're right. Sorry I did not think of that. Closing this.